src/Profile/ContestProfile.cpp

```diff
@@ -29,7 +29,8 @@ Copyright_License {
 void
 Profile::Load(const ProfileMap &map, ContestSettings &settings)
 {
-  if (map.GetEnum(ProfileKeys::OLCRules, settings.contest)) {
+  if (map.GetEnum(ProfileKeys::OLCRules, settings.contest) || 
```

Comment:
This means that the old config always wins, i.e. changing the setting never has an effect.
But let's not rename the ProfileKey for now. This is error prone as you see; let's concentrate on renaming stuff in the code, things that can't break any code.

Reply:
I agree, I had some reluctance there as well. I just left a note in the ProfileKey now.